config.go

```diff
@@ -318,6 +340,29 @@ func (c *compiledModule) ExportedMemories() map[string]api.MemoryDefinition {
 	return c.module.ExportedMemories()
 }
 
+// CustomSections implements CompiledModule.CustomSections
+func (c *compiledModule) CustomSections() []api.CustomSection {
+	var ret []api.CustomSection
+	for _, d := range c.module.CustomSections {
+		dd := &customSection{data: d.Data, name: d.Name}
+		ret = append(ret, dd)
```

Comment:
Let's use `make` to pre-reserve the memory so that we can avoid unnecessary memory allocation in `append` inside the loop. (whenever possible doing it is the norm in Go! 🕵️)

```suggestion
	ret := make([]api.CustomSection, len(c.module.CustomSections))
	for i, d := range c.module.CustomSections {
		ret[i] = &customSection{data: d.Data, name: d.Name}
```

Reply:
FYI: this slightly alters the semantics, it will no longer return nil but an empty slice (which is fine, just FYI because I had to update `Test_compiledModule_CustomSections`!)